WalletWasabi/Crypto/ScalarVector.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin.Secp256k1;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto
+{
+	public class ScalarVector : IEnumerable<Scalar>
+	{
+		public ScalarVector(IEnumerable<Scalar> scalars)
+		{
+			Guard.NotNullOrEmpty(nameof(scalars), scalars);
+			Scalars = scalars.ToArray();
+		}
+
+		private IEnumerable<Scalar> Scalars { get; }
+
+		public IEnumerator<Scalar> GetEnumerator() =>
+			Scalars.GetEnumerator();
+
+		public int Count => Scalars.Count();
+
+		public static GroupElement operator *(ScalarVector scalars, GroupElementVector groupElements)
+		{
+			Guard.NotNull(nameof(scalars), scalars);
+			Guard.NotNull(nameof(groupElements), groupElements);
```

Comment:
I think it's unnecessary as we added nullmagic, so notnulling should be ensured on the compiler warning level.

Reply:
I've been thinking a lot about that and also googling because I still don't know if we are understanding it 100% right. The fact that the compiler alerts you about nulls doesn't make impossible to pass nulls. That's why in c# 9 there is a new operator called "simplified parameter null validation code" designed to make the parameters null validation easier. So, it seems to me that null validation is still needed. In the worst case it is probably redundant.